src/validations/trip.validation.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable import/prefer-default-export */
+import Joi from 'joi';
+import ResponseService from '../services/response.service';
+
+export const validateReturnTrip = (req, res, next) => {
+	const schema = Joi.object(
+
+		{
+			departureFrom: Joi.string().trim().strict()
+				.min(3)
+				.max(30)
+				.required()
+				.messages({
+					'string.empty': 'departureFrom is not allowed to be empty',
+					'string.min': 'departureFrom length must be at least 3 characters long',
+					'string.max': ' departureFrom length must be less than or equal to 30 characters long',
+					'any.required': 'departureFrom is required',
+					'string.trim': 'departureFrom must not have white spces at the beginning and at the end'
+				}),
+			departureTo: Joi.string().trim().strict()
+				.min(3)
+				.max(30)
+				.required()
+				.messages({
+					'string.empty': 'departureTo is not allowed to be empty',
+					'string.min': 'departureTo length must be at least 3 characters long',
+					'string.max': 'departureTo length must be less than or equal to 30 characters long',
+					'any.required': 'departureTo is required',
+					'string.trim': 'departureTo must not have white spces at the beginning and at the end'
+				}),
+			startingDate: Joi.string()
+				.regex(/^\d{4}\-(0[1-9]|1[012])\-(0[1-9]|[12][0-9]|3[01])$/)
+				.required()
+				.messages({
+					'string.pattern.base': 'birthdate shoul be in the format (yyyy-mm-dd) example:(2017-11-30)',
+					'any.required': 'birthdate is required',
+					'string.empty': 'birthdate is not allowed to be empty',
+				}),
+			returningDate: Joi.string()
+				.regex(/^\d{4}\-(0[1-9]|1[012])\-(0[1-9]|[12][0-9]|3[01])$/)
```

Comment:
In this file ask yourself this question. What can happen if a user enter a returning date less than starting date that would be a disaster